### PR TITLE
Replace BranchAndBuildFileWriterTest with initializer

### PR DESCRIPTION
**Pull request by ideco**
This does not solve the problem, that the build.xml should be created at the end of the run (see issue #1) but seems less confusing than having a dummy test.

### CODE REVIEW

File: test/src/test/java/org/scenarioo/pizza/scenarioo/ScenariooBuildContext.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.Date;
+
+
+public class ScenariooBuildContext {
+
+    private static final String BRANCH_NAME = "pizza-delivery-" + envOrDefault("BRANCH_NAME", "master");
+    private static final String BUILD_NAME = envOrDefault("BUILD_NUMBER", "undefined");
```
Comment (bruderol) [inline test/src/test/java/org/scenarioo/pizza/scenarioo/ScenariooBuildContext.java]: I would like to see "build-" as prefix in front of the build name (just a number is a little bit strange maybe)

File: test/src/test/java/org/scenarioo/pizza/scenarioo/ScenariooBuildContext.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    private static Build createBuild(String name) {
+        Build build = new Build();
+        build.setName(name);
+        build.setDate(getCurrentDate());
+        build.setRevision(envOrDefault("GIT_COMMIT", "undefined"));
+        return build;
+    }
+
+    private static String envOrDefault(String name, String defaultValue) {
```
Comment (bruderol) [inline test/src/test/java/org/scenarioo/pizza/scenarioo/ScenariooBuildContext.java]: I think `getEnvironmentVariableOrDefault` would be better understandable.

File: test/src/test/java/org/scenarioo/pizza/scenarioo/ScenariooBuildContext.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        String environmentVariable = System.getenv(name);
+        return environmentVariable != null ? environmentVariable : defaultValue;
+    }
+
+    private static Date getCurrentDate() {
+        ZonedDateTime zonedDateTime = LocalDateTime.now().atZone(ZoneId.systemDefault());
```
Comment (bruderol) [inline test/src/test/java/org/scenarioo/pizza/scenarioo/ScenariooBuildContext.java]: isnt LocaleDateTime.now() sufficient?

File: test/src/test/java/org/scenarioo/pizza/scenarioo/ScenariooBuildContext.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    private static final File DOCUMENTATION_ROOT = new File("build/scenariooDocumentation");
+
+    static {
+        DOCUMENTATION_ROOT.mkdirs();    // Create folder for generated documentation
+        ScenarioDocuWriter writer = getNewWriter();
+        writer.saveBranchDescription(createBranch(BRANCH_NAME));    // Create 'branch.xml'
+        writer.saveBuildDescription(createBuild(BUILD_NAME));       // Create 'build.xml'
+        writer.flush();
+    }
+
```
Comment (bruderol) [inline test/src/test/java/org/scenarioo/pizza/scenarioo/ScenariooBuildContext.java]: I would recommend to change this, as follows:

```
static {
           // on startup - once per test run:
           initScenariooReportForCurrentBuildRun();
 }

/**
 * Create directory to store the scenarioo report files inside
 * and store the needed description files for the branch and the build.
 * This has to be done at least once for a full scenarioo report build.
 */
private void initScenariooReportForCurrentBuildRun() {
           DOCUMENTATION_ROOT.mkdirs();
           ScenarioDocuWriter writer = getNewWriter();
           writer.saveBranchDescription(createBranch(BRANCH_NAME));
           writer.saveBuildDescription(createBuild(BUILD_NAME));
           writer.flush();
}
```
Reasons:
 1. better explanations
 2. I do not like inline comments behind code, but rather good method names and some good explanations of whole blocks/methods instead.
 3. I think `// Create branch.xml` is too technical and not realy important (why should the user care that this creates a XML ? could as well be a JSON later. the name of the method allready says all that he needs to know kind of).

**Comment by bruderol** [comment]
@ideco rest looks great - thanks! what do you think? can you change this and then merge it?

File: test/src/test/java/org/scenarioo/pizza/scenarioo/ScenariooBuildContext.java (hunk 4 above)
Comment (bruderol) [reply test/src/test/java/org/scenarioo/pizza/scenarioo/ScenariooBuildContext.java]: @ideco I changed my proposal above again  -what do you think? Makes sense like that?

File: test/src/test/java/org/scenarioo/pizza/scenarioo/ScenariooBuildContext.java (hunk 3 above)
Comment (ideco) [reply test/src/test/java/org/scenarioo/pizza/scenarioo/ScenariooBuildContext.java]: This is the standard way to convert LocalDateTime to java.util.Date. However, we can simplify ...

File: test/src/test/java/org/scenarioo/pizza/scenarioo/ScenariooBuildContext.java (hunk 2 above)
Comment (ideco) [reply test/src/test/java/org/scenarioo/pizza/scenarioo/ScenariooBuildContext.java]: Agree

File: test/src/test/java/org/scenarioo/pizza/scenarioo/ScenariooBuildContext.java (hunk 1 above)
Comment (ideco) [reply test/src/test/java/org/scenarioo/pizza/scenarioo/ScenariooBuildContext.java]: OK

File: test/src/test/java/org/scenarioo/pizza/scenarioo/ScenariooBuildContext.java (hunk 4 above)
Comment (ideco) [reply test/src/test/java/org/scenarioo/pizza/scenarioo/ScenariooBuildContext.java]: Absolutely!

**Comment by bruderol** [comment]
@ideco cool! But your last commit was done with unrecognized email address - please always configure your git correctly before commiting - and: register your business email as a secondary email address on your github account to get the commit correctly assigned.

**Pull request merged**